### PR TITLE
Added Container Registry storage

### PR DESCRIPTION
**Pull request by phillip-kruger**
This PR change the Deployer in hacbs-sidecar to load the deployer from a factory and then have the ability to add more deployer.
The current default is still S3 for now, but this PR also adds a Container Registry Storage option.

Signed-off-by: Phillip Kruger <phillip.kruger@gmail.com>

### CODE REVIEW

File: java-components/sidecar/src/main/java/com/redhat/hacbs/sidecar/resources/deploy/containerregistry/Gav.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
+package com.redhat.hacbs.sidecar.resources.deploy.containerregistry;
+
+import java.util.Objects;
+
+public class Gav {
+    private String groupId;
```
Comment (stuartwdouglas) [inline java-components/sidecar/src/main/java/com/redhat/hacbs/sidecar/resources/deploy/containerregistry/Gav.java:6]: I know it does not really matter here but these should really be immutable. In general mutability + equals/hashCode can result in lots of problems, as the hash code would change if you modify the object.

File: java-components/sidecar/src/main/java/com/redhat/hacbs/sidecar/resources/deploy/containerregistry/ImageData.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+import java.nio.file.Path;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class ImageData {
+
+    private Path artifactsPath;
+    private Set<Gav> gavs;
```
Comment (stuartwdouglas) [inline java-components/sidecar/src/main/java/com/redhat/hacbs/sidecar/resources/deploy/containerregistry/ImageData.java:11]: Same here.

**Review by stuartwdouglas — CHANGES_REQUESTED** [review]
Looks good, just a few minor things. I will hold of on merging it though until we have stability in the e2e tests.

**Comment by gabemontero** [comment]
with https://github.com/redhat-appstudio/jvm-build-service/pull/129 in repo e2e's are green

/retest

**Comment by gabemontero** [comment]
though the compile is failing now at least in https://github.com/redhat-appstudio/jvm-build-service/runs/7493797597?check_suite_focus=true

**Comment by phillip-kruger** [comment]
@stuartwdouglas seems like the error has something to do with testcontainers that needs ryuk to performs fail-safe cleanup of containers. see https://www.testcontainers.org/features/configuration/

How does other builds that needs testcontainers handle this ?

**Comment by stuartwdouglas** [comment]
What makes you say that? It looked like docker.io failed to service an image to me.

**Review by stuartwdouglas — APPROVED** [review]

**Pull request merged**